File: Assignment/Sukendu Dutta/Day-14(24-07-2023)Project1/CurrencyCalc.java
import java.util.Scanner;
public class CurrencyConverter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Currency Conversion Options:");
        System.out.println("1. USD to INR");
        System.out.println("2. USD to EUR");
        System.out.println("3. USD to KWD");
        System.out.println("4. INR to USD");
        System.out.println("5. EUR to USD");
        System.out.println("6. KWD to USD");
        System.out.print("Enter your choice (1-6): ");
        int choice = scanner.nextInt();
        System.out.print("Enter the amount: ");
        double amount = scanner.nextDouble();
        double convertedAmount = 0.0;
        String fromCurrency = "";
        String toCurrency = "";
        switch (choice) {
            case 1:
                convertedAmount = usdToInr(amount);
                fromCurrency = "USD";
                toCurrency = "INR";
                break;
            case 2:
                convertedAmount = usdToEur(amount);
                fromCurrency = "USD";
                toCurrency = "EUR";
                break;
            case 3:
                convertedAmount = usdToKwd(amount);
                fromCurrency = "USD";
                toCurrency = "KWD";
                break;
            case 4:
                convertedAmount = inrToUsd(amount);
                fromCurrency = "INR";
                toCurrency = "USD";
                break;
            case 5:
                convertedAmount = eurToUsd(amount);
                fromCurrency = "EUR";
                toCurrency = "USD";
                break;
            case 6:
                convertedAmount = kwdToUsd(amount);
                fromCurrency = "KWD";
                toCurrency = "USD";
                break;
            default:
                System.out.println("Invalid choice. Please enter a number from 1 to 6.");
                System.exit(1);
        }
        System.out.println(amount + " " + fromCurrency + " is equal to " + convertedAmount + " " + toCurrency + ".");
        scanner.close();
    }
    public static double usdToInr(double usd) {
        return usd * 74.5;
    }
    public static double usdToEur(double usd) {
        return usd * 0.85;
    }
    public static double usdToKwd(double usd) {
        return usd * 0.3;
    }
    public static double inrToUsd(double inr) {
        return inr / 74.5;
    }
    public static double eurToUsd(double eur) {
        return eur / 0.85;
    }
    public static double kwdToUsd(double kwd) {
        return kwd / 0.3;
    }
}

File: Assignment/Gourab Kumar Sahoo/Day-14(24-07-2023)/CurrencyCalc.py
def convert_currency(amount, from_currency, to_currency, exchange_rates):
    if from_currency not in exchange_rates or to_currency not in exchange_rates:
        print("Currency not supported.")
        return None

    from_rate = exchange_rates[from_currency]
    to_rate = exchange_rates[to_currency]

    converted_amount = amount * (to_rate / from_rate)
    return converted_amount

if __name__ == "__main__":
    print("Welcome to the Currency Calculator!")

    exchange_rates = {
        "USD": 1.0,    # Example: 1 USD = 1.0 USD (base currency)
        "EUR": 0.85,   # Example: 1 USD = 0.85 EUR
        "GBP": 0.75,   # Example: 1 USD = 0.75 GBP
        "JPY": 110.0,  # Example: 1 USD = 110 JPY
        # Add more exchange rates as needed
    }

    while True:
        print("\nChoose an option:")
        print("1. Convert from USD")
        print("2. Convert to USD")
        print("3. Exit")

        choice = input("Enter your choice (1-3): ")

        if choice == '1':
            amount = float(input("Enter amount in USD: "))
            to_currency = input("Enter target currency (e.g., EUR, GBP, JPY): ").upper()
            converted_amount = convert_currency(amount, "USD", to_currency, exchange_rates)
            if converted_amount is not None:
                print(f"{amount:.2f} USD is equal to {converted_amount:.2f} {to_currency}.")

        elif choice == '2':
            amount = float(input("Enter amount: "))
            from_currency = input("Enter source currency (e.g., EUR, GBP, JPY): ").upper()
            converted_amount = convert_currency(amount, from_currency, "USD", exchange_rates)
            if converted_amount is not None:
                print(f"{amount:.2f} {from_currency} is equal to {converted_amount:.2f} USD.")

        elif choice == '3':
            print("Thank you for using the Currency Calculator. Goodbye!")
            break

        else:
            print("Invalid choice. Please choose a valid option (1-3).")
